### PR TITLE
Reformatted source code (collar and spares)

### PR DESCRIPTION
**Pull request by riej**
Replacement of old PR, but with latest changes.
Re-formatted all collar and spares scripts. Not changed yet: AO, cuffs, installer, remote.
Plus wrote simple code style guide, based on my changes.

Please look and test everything.

### CODE REVIEW

**Comment by riej** [comment]
@nirea @zontreck can you look it please?

File: CODE_STYLING.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+Use `TRUE` and `FALSE` for boolean purposes.
+Use predefined constants instead of "magic numbers" (for example, `llAttachToAvatar(ATTACH_NECK);` instead of `llAttachToAvatar(39);`).
+Split different code blocks or steps with spaces.
+Do not make unnecessary complexity.
+Avoid of uncommon hacks (for example, use `llGetListLength(lMyList)` instead of `(lMyList != [])`).
+Avoid of unnecessary memory savings, like loop variables in global scope.
+Feed unicorns, not fire dragons.
+
+### Indentation
+Use spaces, not tabs. Indent size is 4 spaces.
```
Comment (zontreck) [inline CODE_STYLING.md]: Tabs are okay though if using the in-world script editor as it converts the tabs to the proper spacing.

File: src/collar/oc_anim.lsl [hunk 2]
```diff
@@ -93,778 +97,785 @@ list g_lMenuIDs; // Three strided list of kAvatar, kDialogID, sMenuName
...
 
-  lButtons += Checkbox(g_bTweakPoseAO, "AntiSlide");
-  lButtons += ["AO Menu", "AO ON", "AO OFF", "Pose"];
+    if (llGetInventoryType("~stiff") == INVENTORY_ANIMATION) {
+        if (g_bPosture) {
+            sPrompt += " and has their neck forced stiff.";
+            lButtons += ["☑ Posture"];
+        } else {
+            sPrompt += " and can relax their neck.";
+            lButtons += ["☐ Posture"];
```
Comment (zontreck) [inline src/collar/oc_anim.lsl]: I like how you moved the posture option into a inventory check. Though i still need to check that this actually works as intended. Will update my review after testing

File: src/collar/oc_com.lsl [hunk 3]
```diff
@@ -566,42 +681,60 @@ default {
...
+            } else {
+                sMessage += "\n" + llList2String(g_lCore5Scripts, i + 1) + "\t(Link# " + llList2String(g_lFoundCore5Scripts, iIndex + 1) + ")";
+            }
+
+            i += 2;
+        } while (i < 10);
+
         i = llGetLinkNumber();
-        if (i != 1) sMessage += "\noc_com\t(not in root prim!)";
+        if (i != 1) {
```
Comment (zontreck) [inline src/collar/oc_com.lsl]: Somehow i feel like this is cleaner... but at the same time, its going to make a lot of scripts a lot longer.

File: src/collar/oc_anim.lsl (hunk 2 above)
Comment (riej) [reply src/collar/oc_anim.lsl]: Ummm, but i haven't moved anything here. Posture option was already inside inventory check, isn't it?

File: src/collar/oc_com.lsl (hunk 3 above)
Comment (riej) [reply src/collar/oc_com.lsl]: Why not, if it makes code well-readable? I suppose the main target is to write good code, not minified.

File: src/collar/oc_anim.lsl (hunk 2 above)
Comment (zontreck) [reply src/collar/oc_anim.lsl]: Ah it looks like you did the way this git diff came out...

File: src/collar/oc_capture.lsl [hunk 4]
```diff
@@ -22,288 +22,359 @@ integer CMD_SAFEWORD = 510;
...
-    }
-    else if (llSubStringIndex(sStrLower,"capture")==0) {
-        if (g_sTempOwnerID != "" && kID==g_kWearer) {
-            llMessageLinked(LINK_DIALOG,NOTIFY,"0"+"%NOACCESS%",g_kWearer);
+    } else if (sStrLower == "capture" || sStrLower == "menu capture") {
+        if  (iNum != CMD_OWNER && iNum != CMD_WEARER) {
+            if (g_bCaptureOn) {
+                Dialog(kID, "\nYou can try to capture %WEARERNAME%.\n\nReady for that?", ["Yes", "No"], [], 0, iNum, "ConfirmCaptureMenu", kID);
+            } else {
+                llMessageLinked(LINK_DIALOG, NOTIFY, "0" + "%NOACCESS%", kID); // Notify(kID, g_sAuthError, FALSE);
```
Comment (zontreck) [inline src/collar/oc_capture.lsl]: After testing your version. I am getting access denied messages in this script. To reproduce. Turn public mode on and also turn on capture.  Then have someone else touch and they'll get a access denied message at least twice.

**Comment by riej** [comment]
I fixed misprints (sorry, made mistake with branches^^), but cannot repeat bug with `oc_capture` (also checked source code and found no differences in logic between old and new variants).

File: src/spares/oc_camera.lsl [hunk 5]
```diff
@@ -152,7 +152,7 @@ string JsonModes() {
     ]);
 }
 
-list JsonModes(string sMode) {
+list JsonModeToList(string sMode) {
```
[the remark this answers is not in the document]
Comment (zontreck) [reply src/spares/oc_camera.lsl]: I get access denied when testing this script on most the options. Please check the code.

File: src/collar/oc_relay.lsl [hunk 6]
```diff
@@ -963,7 +963,7 @@ default {
 
     listen(integer iChannel, string sWho, key kID, string sMsg) {
         if (iChannel == SAFETY_CHANNEL) {
-            llMessageLinked(LINK_DIALOG, NOTIFY, "0\n\n⚠ " + sWho + " detected ⚠\n\nTo prevent conflicts this relay is being detached now! If you wish to use "+who+" anyway, type \"/%CHANNEL% %PREFIX% relay off\" to temporarily disable or type \"/%CHANNEL% %PREFIX% rm relay\" to permanently uninstall the relay plugin.\n", g_kWearer);
+            llMessageLinked(LINK_DIALOG, NOTIFY, "0\n\n⚠ " + sWho + " detected ⚠\n\nTo prevent conflicts this relay is being detached now! If you wish to use " + sWho + " anyway, type \"/%CHANNEL% %PREFIX% relay off\" to temporarily disable or type \"/%CHANNEL% %PREFIX% rm relay\" to permanently uninstall the relay plugin.\n", g_kWearer);
```
[the remark this answers is not in the document]
Comment (zontreck) [reply src/collar/oc_relay.lsl]: Thank you. This script works

**Comment by zontreck** [comment]
oc_sys still needs to be fixed as it had a problematic line preventing compile.

**Comment by zontreck** [comment]
Oh nevermind noticed you fixed oc_sys. ignore my last comment. But if camera and capture can't be reproduced/fixed can you not include those in this PR, fix the issue then include them in a new PR later?

**Comment by zontreck** [comment]
For your code styling guide: There is already a plugin guide actually that just does not include the style information. I suggest going off of that document and changing it to add a style section. https://github.com/OpenCollarTeam/opencollarteam.github.io/blob/master/docs/Plugin-Guide.md

**Comment by zontreck** [comment]
Hey there has not been any progress on this in awhile. And as this pull request is not in a stage where it can be merged.. I will close it.

**Pull request closed without merging**